The Titan Insurance Company has just installed a new incentive payment scheme for its lift policy sales force. It wants to have an early view of the success or failure of the new scheme. Indications are that the sales force is selling more policies, but sales always vary in an unpredictable pattern from month to month and it is not clear that the scheme has made a significant difference.

Life Insurance companies typically measure the monthly output of a salesperson as the total sum assured for the policies sold by that person during the month. For example, suppose salesperson X has, in the month, sold seven policies for which the sums assured are £1000, £2500, £3000, £5000, £10000, £35000. X's output for the month is the total of these sums assured, £61,500. Titan's new scheme is that the sales force receives low regular salaries but are paid large bonuses related to their output (i.e. to the total sum assured of policies sold by them). The scheme is expensive for the company, but they are looking for sales increases which more than compensate. The agreement with the sales force is that if the scheme does not at least break even for the company, it will be abandoned after six months.

The scheme has now been in operation for four months. It has settled down after fluctuations in the first two months due to the changeover.

To test the effectiveness of the scheme, Titan have taken a random sample of 30 salespeople measured their output in the penultimate month prior to changeover and then measured it in the fourth month after the changeover (they have deliberately chosen months not too close to the changeover). The outputs of the salespeople are shown in Table 1

 

Questions
1. Find the mean of old scheme and new scheme column. (5 points)

2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)
3. What conclusion does the test (p-value) lead to? (2.5 points)
4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:
        a) The probability of a type 1 error? (2.5 points)

        b) What is the p- value of the hypothesis test if we test for a difference of $5000? (5 points)

        c) Power of the test (5 points)

In [1]:
import numpy as np
import pandas as pd
from scipy.stats import ttest_1samp, ttest_ind, ttest_rel, t
import matplotlib.pyplot as plt
from math import sqrt

%matplotlib inline


In [2]:
## Reading a dataset from a csv file using pandas.
df = pd.read_csv('data.csv')
df.head(3)

,SALESPERSON,Old Scheme (in thousands),New Scheme (in thousands)
0,1,57,62
1,2,103,122
2,3,59,54


In [3]:

df.describe()

,SALESPERSON,Old Scheme (in thousands),New Scheme (in thousands)
count,30.000000,30.000000,30.000000
mean,15.500000,68.033333,72.033333
std,8.803408,20.455980,24.062395
min,1.000000,28.000000,32.000000
25%,8.250000,54.000000,55.000000
50%,15.500000,67.000000,74.000000
75%,22.750000,81.500000,85.750000
max,30.000000,110.000000,122.000000


# 1. Find the mean of old scheme and new scheme column. (5 points)

In [4]:
print("Mean of Old Scheme = ", df['Old Scheme (in thousands)'].mean())
print("Mean of New Scheme = ", df['New Scheme (in thousands)'].mean())

Mean of Old Scheme =  68.03333333333333
Mean of New Scheme =  72.03333333333333


# 2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)

In [5]:
#The level of significance (Alpha ) = 0.05.
alpha = 0.05

scheme1 = df['Old Scheme (in thousands)'] * 1000
scheme2 = df['New Scheme (in thousands)'] * 1000

t_statistic, p_value = ttest_1samp(scheme1 - scheme2, 0)
print(t_statistic, p_value)

t_statistic, p_value = ttest_rel(scheme1, scheme2)
print(t_statistic, p_value)

#since this is a 2 tailed test, need to divide pvalue by 2 to get onetailed result
p_value = p_value / 2
print ("The p value =", p_value)

-1.5559143823544377 0.13057553961337662
-1.5559143823544377 0.13057553961337662
The p value = 0.06528776980668831


## Answer

Since p-value = 0.06528776980668831 is higher than 0.05, we accept (fail to reject) NULL hypothesis. The new scheme has NOT significantly raised outputs.


# 3. What conclusion does the test (p-value) lead to? (2.5 points)

## Answer
The p value is 0.06528776980668831 which is greater than the 0.05. So **ACCEPT the null hypothesis** that the new scheme did not raise the output significantly. 

# 4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:
## a) The probability of a type 1 error? (2.5 points)


## Answer
Probability of Type I error which is predetermined by significance level. Since the significance level is 0.05, then type I error is 0.05. i.e. 5% probability we make Type I error – rejecting null hypothesis when it is true.


## b) What is the p- value of the hypothesis test if we test for a difference of $5000? (5 points)

   

In [6]:
# H0: μd ≤ 5000  
# HA: μd > 5000

In [7]:
#Number of items
n =  df.shape[0]
print("Count = ", n)

#Standard Deviation
sd = (df['Old Scheme (in thousands)'] - df['New Scheme (in thousands)']).std()
print("Standard deviation = ", sd)

#Degree of Freedom
DF = n-1
print("Degree of Freedom = ", DF)

#With α = 0.05 and df = 29, critical value for t statistic (or t_critical ) will be   1.699127.
t_critical = t.ppf(1-alpha, DF)
print("t_critical = ", t_critical)

# t_critical = (diff_in_mean - uD) / sd / sqrt(n)
# t_critical = (diff_in_mean - 0) / sd / sqrt(n)
diff_in_mean = t_critical / sqrt(n) * sd
print("diff_in_mean = ", diff_in_mean)

#tstat = (d^ - uD) / Sd / sqrt(n)
uD = 5000 / 1000 #5 same unit as dataframe
tstat = (diff_in_mean - uD) /( sd / sqrt(n))
print("tstat = ", tstat)

#calculate the p-value
p_val = t.sf(np.abs(tstat), n-1)
print("p_val = ", p_val)

Count =  30
Standard deviation =  14.08104619937615
Degree of Freedom =  29
t_critical =  1.6991270265334972
diff_in_mean =  4.368176156228719
tstat =  -0.2457659514095499
p_val =  0.4037972524794316


## c) Power of the test (5 points)

In [8]:
#Power of the hypothesis test is 1 minus the probability of Type II error. Basically the power of a test is the probability that we make the right decision when the null is not correct (i.e. we correctly reject it) Power of test is 1- beta = 1-0.40379 = 0.5962
beta = p_val
power_of_test = 1 - beta
print("Power of Test = ", round(power_of_test * 100, 2), "%")

Power of Test =  59.62 %
